Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -428,6 +324,108 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
 
         #endregion
 
+        #region Private Helpers
+
+        private IProjectEntry RemoveEntry(Uri documentUri) {
+            var entry = _projectFiles.RemoveEntry(documentUri);
+            _openFiles.Remove(documentUri);
+            return entry;
+        }
+
+        private async Task DoInitializeAsync(InitializeParams @params) {
+            _analyzer = await CreateAnalyzer(@params.initializationOptions.interpreter);
+
+            _clientCaps = @params.capabilities;
+            _settings.SetCompletionTimeout(_clientCaps?.python?.completionsTimeout);
+            _traceLogging = _clientCaps?.python?.traceLogging ?? false;
+            _analyzer.EnableDiagnostics = _clientCaps?.python?.liveLinting ?? false;
+
+            _reloadModulesQueueItem = new ReloadModulesQueueItem(_analyzer);
+
+            if (@params.initializationOptions.displayOptions != null) {
+                DisplayOptions = @params.initializationOptions.displayOptions;
+            }
+            _displayTextBuilder = DocumentationBuilder.Create(DisplayOptions);
+
+            if (string.IsNullOrEmpty(_analyzer.InterpreterFactory?.Configuration?.InterpreterPath)) {
+                LogMessage(MessageType.Log, "Initializing for generic interpreter");
+            } else {
+                LogMessage(MessageType.Log, $"Initializing for {_analyzer.InterpreterFactory.Configuration.InterpreterPath}");
+            }
+
+            if (@params.rootUri != null) {
+                _rootDir = @params.rootUri.ToAbsolutePath();
+            } else if (!string.IsNullOrEmpty(@params.rootPath)) {
+                _rootDir = PathUtils.NormalizePath(@params.rootPath);
+            }
+
+            SetSearchPaths(@params.initializationOptions.searchPaths);
+
+            if (_rootDir != null && !(_clientCaps?.python?.manualFileLoad ?? false)) {
+                LogMessage(MessageType.Log, $"Loading files from {_rootDir}");
+                _loadingFromDirectory = LoadFromDirectoryAsync(_rootDir);
+            }
+        }
+
+        private T ActivateObject<T>(string assemblyName, string typeName, Dictionary<string, object> properties) {
+            if (string.IsNullOrEmpty(assemblyName) || string.IsNullOrEmpty(typeName)) {
+                return default(T);
+            }
+            try {
+                var assembly = File.Exists(assemblyName) ? AssemblyName.GetAssemblyName(assemblyName) : new AssemblyName(assemblyName);
+                var type = Assembly.Load(assembly).GetType(typeName, true);
```

Comment:
Is it going to be for the public API, or is it for internal purposes only?

Reply:
Yes, public/stable API

Reply:
Is design finalized or we can discuss it?

Reply:
We can discuss it, but consistency with the existing public API is going to be a strong argument in favour of this design :)